Mage/Archivist/Archivist.cs

```diff
@@ -0,0 +1,268 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+using Newtonsoft.Json.Linq;
+
+public class Archivist : EventEmitter<JObject> {
+	private Mage mage { get { return Mage.instance; } }
+	private Logger logger { get { return mage.logger("archivist"); } }
+
+	// Local cache of all retrieved vault values
+	private Dictionary<string, VaultValue> _cache = new Dictionary<string, VaultValue>();
+
+
+	// Constructor
+	public Archivist () {
+		// Apply changes to cached vault values when applyDiff event is fired
+		mage.eventManager.on ("archivist:applyDiff", (object sender, JToken diff) => {
+			string topic = diff["key"]["topic"].ToString();
+			JObject index = (JObject)diff["key"]["index"];
```

Comment:
Here I see typecasting, but I don't see it one line above...? Seems needless then.